src/platform/tls_schannel.c

```diff
@@ -3045,19 +3163,44 @@ QuicHpComputeMask(
     )
 {
     ULONG TempSize = 0;
-    QUIC_STATUS Status =
-        NtStatusToQuicStatus(
-        BCryptEncrypt(
-            (BCRYPT_KEY_HANDLE)Key,
-            (uint8_t*)Cipher,
-            QUIC_HP_SAMPLE_LENGTH * BatchSize,
-            NULL,
-            NULL,
-            0,
-            Mask,
-            QUIC_HP_SAMPLE_LENGTH * BatchSize,
-            &TempSize,
-            0));
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+    if (Key->Aead == QUIC_AEAD_CHACHA20_POLY1305) {
+        uint8_t Zero[5] = {0};
+        Key->Info->cbNonce = QUIC_HP_SAMPLE_LENGTH;
+        for (uint32_t i = 0, Offset = 0; i < BatchSize; ++i, Offset += QUIC_HP_SAMPLE_LENGTH) {
```

Comment:
Is there no way to do the batch in a single call?

Reply:
No. ChaCha20 is a streaming cipher, not a block cipher. The QUIC-TLS spec uses it to get 5 bytes of the stream, for a given unique IV (the packet sample). Because each header uses a different IV, it produces a different stream per packet. Which makes it not possible to do a single call.